packages/nitro-protocol/src/contract/challenge.ts

```diff
@@ -1,9 +1,50 @@
 import {Bytes32} from './types';
-import {keccak256, defaultAbiCoder} from 'ethers/utils';
-import {State, hashState} from './state';
+import {keccak256, defaultAbiCoder, bigNumberify} from 'ethers/utils';
+import {hashState, VariablePart, State} from './state';
+import {Channel} from '..';
+import {decodeOutcome} from './outcome';
 
 export function hashChallengeMessage(challengeState: State): Bytes32 {
   return keccak256(
     defaultAbiCoder.encode(['bytes32', 'string'], [hashState(challengeState), 'forceMove']),
   );
 }
+
+export interface ChallengeRegisteredEvent {
+  challengerAddress: string;
+  finalizesAt: string;
+  challengeStates: State[];
+}
+export function getChallengeRegisteredEvent(eventArgs): ChallengeRegisteredEvent {
+  // TODO: There must be a better way of parsing out the event
```

Comment:
`event.args` is an array with properties. So, something like
```
const { turnNumRecord, finalizesAt } = eventArgs
```
should work, as long as `eventArgs === event.args` for some `event` fetched by ethers.

Reply:
It looks like the last argument passed into the listener from `NitroAdjudicator.on() is the the actual event, so we could pull the `args` off from there.  Still slightly annoying that we have to manually parse `FixedPart`/`VariablePart`